### PR TITLE
xapi-guard: Rename varstored modules

### PR DESCRIPTION
**Pull request by psafont**
Varstored is not the only daemon using xapi-guard. While the name of the executable can't be easily changed, the modules in it can be changed to reflect their current and immediate future usage:

The privileged socket between xenopsd and xapi-guard are independent of varstore. and the deprivileged socket used by varstored (and soon swtpm) are served by a xapi-guard server. The new names should better reflect these functionality and use-cases

### CODE REVIEW

File: ocaml/xenopsd/lib/xenops_sandbox.ml [hunk 1]
```diff
@@ -219,11 +219,11 @@ module Swtpm : GUARD = struct
      is needed to /dev/urandom *)
   let base_directory = "/var/lib/xcp/run"
 
-  let create dbg ~vm_uuid ~domid ~path =
-    Varstore_privileged_client.Client.create dbg vm_uuid domid path
+  let create dbg ~vm_uuid ~gid ~path =
+    Varstore_privileged_client.Client.create dbg vm_uuid gid path
```
Comment (edwintorok) [inline ocaml/xenopsd/lib/xenops_sandbox.ml]: Do we need to update xapi-idl if it mentions domid?

**Review by edwintorok — COMMENTED** [review]
Nice cleanups/refactoring of xapi-guard.

**Review by lindig — APPROVED** [review]

File: ocaml/xenopsd/lib/xenops_sandbox.ml [hunk 2]
```diff
@@ -168,6 +171,14 @@ module Guard (G : GUARD) : SANDBOX = struct
     (chroot, Chroot.chroot_path_inside socket_path)
 
   let create ~domid ~vm_uuid path =
+    ( match chroot ~domid ~vm_uuid with
+    | chroot when Sys.file_exists chroot.root ->
+        D.info "%s cleaning up existing chroot for domain %i" __FUNCTION__ domid ;
+        Chroot.destroy chroot
```
[the remark this answers is not in the document]
Comment (psafont) [reply ocaml/xenopsd/lib/xenops_sandbox.ml]: I don't we care about this recovery anymore as we don't support an upgrade path from the old backend to the new one

**Comment by psafont** [comment]
Rebasing the PR leads to a higher amount of lines to be removed and a lower amount of introduces ones :)

File: ocaml/xenopsd/lib/xenops_sandbox.ml (hunk 2 above)
[the remark this answers is not in the document]
Comment (psafont) [reply ocaml/xenopsd/lib/xenops_sandbox.ml]: I've pulled this change out while we decide what to do with upgrades

**Review by edwintorok — APPROVED** [review]

**Pull request merged**